src/ngx_http_lua_shdict.c

```diff
@@ -1257,24 +1257,50 @@ ngx_http_lua_shdict_incr(lua_State *L)
     ngx_http_lua_shdict_node_t  *sd;
     double                       num;
     double                       init = 0;
+    int                          has_init = 0;
     u_char                      *p;
     ngx_shm_zone_t              *zone;
     double                       value;
+    lua_Number                   exptime = -1;
     ngx_rbtree_node_t           *node;
                          /* indicates whether to foricibly override other
                           * valid entries */
     int                          forcible = 0;
     ngx_queue_t                 *queue, *q;
+    ngx_time_t                  *tp;
 
     n = lua_gettop(L);
 
-    if (n != 3 && n != 4) {
-        return luaL_error(L, "expecting 3 or 4 arguments, but only seen %d", n);
+    if (n < 3) {
+        return luaL_error(L, "expecting at least 3 arguments, but only seen %d", n);
```

Comment:
This line exceeds 80 columns. Please use the [ngx-releng](https://github.com/openresty/openresty-devel-utils/blob/master/ngx-releng)  tool to check such things.